api-server/src/server/boot/user.js

```diff
@@ -468,6 +471,93 @@ function postResetProgress(req, res, next) {
   );
 }
 
+function createPostResetProjectProgress(app) {
+  return async function postResetProjectProgress(req, res, next) {
+    const { User } = app.models;
+    const userId = req.user.id;
+    const { blocks } = req.body;
+
+    let toResetChallengeIds = [];
+    let undefinedBlocks = [];
+
+    let isSendBlocks = blocks && blocks.length > 0;
+    let isHasValidBlockToReset = false;
+    let isResetSuccessful = false;
+
+    if (isSendBlocks) {
+      for (const dashedBlockName of blocks) {
+        const blockChallengeIds = getChallengeIdsForBlock(dashedBlockName);
+        if (blockChallengeIds != undefined) {
+          toResetChallengeIds.push(...blockChallengeIds);
+          isHasValidBlockToReset = true;
+        } else {
+          undefinedBlocks.push(dashedBlockName);
+        }
+      }
+
```

Comment:
```suggestion

isHasValidBlockToReset = toResetChallengeIds.length > 0;

```

Reply:
Appreciate the suggestion on improving code's readability and logic, @ShaunSHamilton! I've given thought to moving the `isHasValidBlockToReset = toResetChallengeIds.length > 0;` condition into the `if` scope as you recommended.

However, we encounter a scope limitation with this approach. The variable `isHasValidBlockToReset` needs to be accessible outside of the `if` scope, specifically when we're constructing the response message later in the function.